flax/linen/stochastic.py

```diff
@@ -15,23 +15,31 @@
 """Stochastic modules.
 """
 
+from typing import Iterable, Optional
+
 from jax import lax
 from jax import random
 import jax.numpy as jnp
 
-from flax.linen.module import Module, compact
+from flax.linen.module import Module, compact, merge_param
 
 
 class Dropout(Module):
   """Create a dropout layer.
 
     Attributes:
       rate: the dropout probability.  (_not_ the keep rate!)
+      broadcast_dims: dimensions that will share the same dropout mask
+      deterministic: if false the inputs are scaled by `1 / (1 - rate)` and
+        masked, whereas if true, no mask is applied and the inputs are returned as
+        is.
   """
   rate: float
+  broadcast_dims: Iterable[int] = ()
+  deterministic: Optional[bool] = None
 
   @compact
-  def __call__(self, inputs, deterministic=False, rng=None, broadcast_dims=()):
+  def __call__(self, inputs, deterministic: Optional[bool] = None, rng=None):
```

Comment:
I have no context here, but is it obviously fine to remove broadcast_dims from the arguments? Maybe worth checking if any users on GitHub or Google are using this argument?

Reply:
Will check before merging